constraint/pkg/client/drivers/llm/driver.go

```diff
@@ -0,0 +1,266 @@
+package llm
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	openai "github.com/sashabaranov/go-openai"
+
+	apiconstraints "github.com/open-policy-agent/frameworks/constraint/pkg/apis/constraints"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/client/drivers"
+	llmSchema "github.com/open-policy-agent/frameworks/constraint/pkg/client/drivers/llm/schema"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/core/templates"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/types"
+	"github.com/open-policy-agent/opa/storage"
+	"github.com/sethvargo/go-retry"
+	flag "github.com/spf13/pflag"
+	"github.com/walles/env"
+	admissionv1 "k8s.io/api/admission/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+const (
+	maxRetries = 10
+	// need minimum of 2023-12-01-preview for JSON mode.
+	azureOpenAIAPIVersion = "2024-03-01-preview"
+	azureOpenAIURL        = "openai.azure.com"
+	systemPrompt          = "You are a policy engine for Kubernetes designed to output JSON. Input will be a policy definition, Kubernetes AdmissionRequest object, and parameters to apply to the policy if applicable. Output JSON should only have a 'decision' field with a boolean value and a 'reason' field with a string value explaining the decision, only if decision is false. Only output valid JSON."
+)
+
+var (
+	openAIAPIURLv1 = "https://api.openai.com/v1"
+
+	openAIDeploymentName = flag.String("openai-deployment-name", env.GetOr("OPENAI_DEPLOYMENT_NAME", env.String, "gpt-3.5-turbo-0301"), "The deployment name used for the model in OpenAI service.")
+	openAIAPIKey         = flag.String("openai-api-key", env.GetOr("OPENAI_API_KEY", env.String, ""), "The API key for the OpenAI service. This is required.")
+	openAIEndpoint       = flag.String("openai-endpoint", env.GetOr("OPENAI_ENDPOINT", env.String, openAIAPIURLv1), "The endpoint for OpenAI service. Defaults to"+openAIAPIURLv1+". Set this to Azure OpenAI Service or OpenAI compatible API endpoint, if needed.")
+)
+
+type Driver struct {
+	prompts map[string]string
+}
+
+var _ drivers.Driver = &Driver{}
+
+type Decision struct {
+	Name       string
+	Constraint *unstructured.Unstructured
+	Decision   bool
+	Reason     string
+}
+
+type ARGetter interface {
+	GetAdmissionRequest() *admissionv1.AdmissionRequest
+}
+
+// Name returns the name of the driver.
+func (d *Driver) Name() string {
+	return llmSchema.Name
+}
+
+func (d *Driver) AddTemplate(_ context.Context, ct *templates.ConstraintTemplate) error {
+	source, err := llmSchema.GetSourceFromTemplate(ct)
+	if err != nil {
+		return err
+	}
+
+	prompt, err := source.GetPrompt()
+	if err != nil {
+		return err
+	}
+	if prompt == "" {
```

Comment:
other than empty string, are there other prompts that we may want to prevent? could any prompts be used to exploit or manipulate the system?